Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+
+When resolving ambiguity of statements inside the body the dialogue statement must be considered the lowest priority by the implementing program.
+For example `<<Fred Move Left>>` could be read as a command or a dialogue statement, it is to be considered a command by the program.
+
+#### White space
+
+It is normal for dialogue to have leading or trailing white space.
+The handling or trimming of trailing white space in dialogue is unspecified.
+The handling or trimming of leading white space in dialogue is unspecified except in the case of options where it must obey the rules for options.
+
+### Commands
+
+Commands are special statements that have no specific output to be shown but are used for passing messages and directions to other parts of the program and to control the flow of the story.
+
+The possible types of commands are:
+
+- flow control
+- jump
+- directions
+- stop
+- set
+
+All commands must start with the `<<` symbol and end with the `>>` symbol.
+Additional required command are unspecified.
+
+#### Directions
+
+_Directions_ are commands for sending messages from the yarn to the rest of the program.
+Implementing programs must not modify the flow of the yarn based on the direction.
+
+Directions can have any text except for the `#`, `{`, or `}` symbols inside of them.
+
+Directions can also have expressions inside of them, however as with dialogue these must be encapsulated by using the `{` and `}` symbols.
+Any expressions inside of a directions command without being encapsulated must be ignored and treated instead as regular text.
+
+```yarn
+<<Fred Move Left 2>>
+<<Unlock Achievement MetSteve>>
+<<Log {$playerName} Died>>
+```
+are examples of directions.
+
+#### Jump
+
+The _jump_ command is how a yarn program can move from one node to another.
+The jump has two components: the keyword and destination.
+The _keyword_ is the text `jump` and comes first in the command.
+
+The _destination_ is the name of the node to load.
+The destination may be any text but must map to the `Title` of a node somewhere else in the project.
+The destination text may be created using the result of an expression, however this must be wrapped inside `{` `}` symbols.
+The expression must resolve to a string value and must be a string that matches the structure of node names.
+
+The behaviour of an implementing program is unspecified when asked to jump to a destination that doesn't match a title in the project.
+
+Once the jump command has been completed the current node must no longer be used, this means any dialogue, options or commands below the jump are to be ignored.
+From that point on the destination nodes contents must instead be run.
+
+`<<jump nodeName>>` is an example of a jump command, `<<jump {$chosenMurderer}>>` is an example of a jump command using an expression to determine the destination node.
+
+#### Stop
+
+The _stop_ command is for halting all progress on the story.
```

Comment:
Perhaps a little pedantic, but it occurs to me "story" hasn't been defined. In other places I think this is fine, but with this particular part of the specification it might be worth finding another way to communicate the intention here (...not that I have a better recommendation tbh)